### DISCUSSION TITLE
[v2] small CheckMenuItemBuilder api inconsistency

### DISCUSSION THREAD

**Discussion by 8** [tauri-apps/tauri#10575] [General]
Hi,

I am trying out the tauri v2 release candidate and just started using the MenuBuilder and friends and have some minor feedback:

The the way that `CheckMenuItemBuilder` is being used is  somewhat different than the other builders, in that you supply the  manager /  apphandle when calling `build()` call instead of `new()` call.

eg. 
```rs
tauri::menu::CheckMenuItemBuilder::new("bla")
              .build(handle)?
```
Vs:

```rs
tauri::menu::SubmenuBuilder::new(handle, "bla")
              .build()?
```
or
```rs
tauri::menu::MenuBuilder::new(handle)
          .build()?
```

Maybe it makes sense to settle on one approach? Just a suggestion.

Take care,
Martin

**Comment by FabianLars** [comment]
@amrbashir

**Comment by fredgig** [comment]
This caught me off guard as well.
